modules/apps/dispatch/dispatch-talend-web/src/main/java/com/liferay/dispatch/talend/web/internal/process/TalendProcessCallable.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dispatch.talend.web.internal.process;
+
+import com.liferay.petra.process.ProcessCallable;
+import com.liferay.petra.process.ProcessException;
+import com.liferay.portal.kernel.util.ArrayUtil;
+
+import java.io.Serializable;
+
+import java.lang.reflect.Method;
+
+/**
+ * @author Igor Beslic
+ */
+public class TalendProcessCallable implements ProcessCallable<Serializable> {
+
+	public TalendProcessCallable(
+		String[] mainMethodArgs, String mainMethodClassFQN) {
+
+		_mainMethodArgs = ArrayUtil.clone(mainMethodArgs);
+		_mainMethodClassFQN = mainMethodClassFQN;
+	}
+
+	@Override
+	public Serializable call() throws ProcessException {
+		ClassLoader classLoader = TalendProcessCallable.class.getClassLoader();
+
+		try {
+			Class<?> talendJobClass = classLoader.loadClass(
+				_mainMethodClassFQN);
+
+			Method talendJobClassMainMethod = talendJobClass.getMethod(
+				"main", String[].class);
+
+			talendJobClassMainMethod.setAccessible(true);
+
+			talendJobClassMainMethod.invoke(
```

Comment:
Hi @shuyangzhou Talend terminates JVM in its own main method - what is quite OK and LocalProcessExecutor handles it correctly.

I wonder what is your opinion on termination of long running Talend job. Talend does not offer graceful shutdown method and I would like to provide shutdown hook in lambda form where I would like to invoke System.exit(0) from TalendProcessCollable.
In com.liferay.portal.search.elasticsearch7.internal.sidecar.ElasticsearchServerUtil that approach is used.

Reply:
For force termination, you need to hold on the "ProcessChannel.getProcessNoticeableFuture()" to call cancel on it. The ES sidecar solution is to workaround the complex internal details from ES, you don't need that.